### PR TITLE
[ELY-2310] Enhance DigestAuthenticationMechanismTest to compute digest values instead of using hard coded values

### PR DESCRIPTION
**Pull request by miador**
Trying to solve [https://issues.redhat.com/browse/ELY-2310]( https://issues.redhat.com/browse/ELY-2310)
According to spec, the A1 value should be `unq(username) ":" unq(realm) ":" passwd`. I've refactored the A1 according to that and since no password needed in `encode` i removed if block.

### CODE REVIEW

**Comment by skyllarr** [comment]
Hello @miador . Thank you for PR! Pls change your commit title to have the issue number at the beginning, Thanks!

File: tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java [hunk 1]
```diff
@@ -28,17 +27,15 @@
...
 import java.security.Security;
 import java.util.HashMap;
 import java.util.Map;
 
-import static org.wildfly.security.http.HttpConstants.CONFIG_REALM;
-import static org.wildfly.security.http.HttpConstants.DIGEST_NAME;
-import static org.wildfly.security.http.HttpConstants.SHA256;
-import static org.wildfly.security.http.HttpConstants.SHA512_256;
-import static org.wildfly.security.http.HttpConstants.UNAUTHORIZED;
+import static org.wildfly.security.http.HttpConstants.*;
```
Comment (skyllarr) [inline tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java]: Just minor, please do not use `*` in imports

File: tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java [hunk 2]
```diff
@@ -19,7 +19,6 @@
 package org.wildfly.security.http.digest;
 
 import mockit.integration.junit4.JMockit;
-
```
Comment (skyllarr) [inline tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java]: @miador Just minor, this space should be added back

File: tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java [hunk 3]
```diff
@@ -75,70 +72,75 @@ public void testRfc2617() throws Exception {
...
-                "                 response=\"6629fae49393a05397450978507c4ef1\",\n" +
-                "                 opaque=\"00000000000000000000000000000000\",\n" +
-                "                 algorithm=MD5"
-        });
-        mechanism.evaluateRequest(request2);
-        Assert.assertEquals(Status.COMPLETE, request2.getResult());
-    }
-
-    @Test
-    public void testRfc2617EncodedQuery() throws Exception {
```
Comment (skyllarr) [inline tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java]: @miador Why is this test method removed?

File: tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java [hunk 4]
```diff
@@ -75,70 +72,75 @@ public void testRfc2617() throws Exception {
...
-                        "                 uri=\"" + path + "\",\n" +
+                        "                 uri=\"/dir/index.html\",\n" +
                         "                 qop=auth,\n" +
                         "                 nc=00000001,\n" +
                         "                 cnonce=\"0a4f113b\",\n" +
-                        "                 response=\"cc3261565007973c9d647333822091ad\",\n" +
+                        "                 response=\"" + computeDigest("/dir/index.html", "dcd98b7102dd2f0e8b11d0f600bfb0c093", "0a4f113b", "00000001") + "\",\n" +
                         "                 opaque=\"00000000000000000000000000000000\",\n" +
                         "                 algorithm=MD5"
-        }, new URI(uri));
```
Comment (skyllarr) [inline tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java:141]: @miador testRfc2617EncodedPath test method was removed, maybe you need to rebase on 1.x?

File: tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java (hunk 3 above)
Comment (miador) [reply tests/base/src/test/java/org/wildfly/security/http/digest/DigestAuthenticationMechanismTest.java]: Oh sorry. Adding back

**Comment by miador** [comment]
The `password` value for SHA256 is with lowecase "o", so the `password` should be `Circle of Life`.
Please see: https://www.rfc-editor.org/errata_search.php?rfc=7616&eid=4495

**Review by skyllarr — APPROVED** [review]

**Comment by skyllarr** [comment]
> The `password` value for SHA256 is with lowecase "o", so the `password` should be `Circle of Life`. Please see: https://www.rfc-editor.org/errata_search.php?rfc=7616&eid=4495

@miador  Currently some tests use uppercase `Of` and some lowercase `of`. It is not important as long as both the server's handler and client use the same. This PR LGTM, thank you.

**Comment by miador** [comment]
Hello,
I am doing a cleanup through my PRs. Will this be merged? If not, i can close.

**Comment by skyllarr** [comment]
@fjuma Please review

**Review by fjuma — APPROVED** [review]
Thanks very much @miador and apologies for the delayed review!

**Pull request merged**